cypress/support/step_definitions/index.ts

```diff
@@ -58,6 +67,28 @@ When('I wait for {int} milliseconds/ms', function (ms: number) {
   cy.wait(ms)
 })
 
+When('I {string} the {string} element', (event: string, selector: string) => {
+  switch (event) {
+    case 'check':
+    case 'click': {
+      $(selector).then(($el) => {
+        const el = $el[0]
+        const parent = el.parentElement
+
+        // Automatically uses `{ force: true }` when trying to activate KInputSwitch as its actual `input` element has `display: none`.
+        const options: Partial<Cypress.ClickOptions> = {}
+        if (parent !== null && parent.classList.contains('k-input-switch')) {
+          options.force = true
+        }
+
+        cy.wrap($el).click(options)
+      })
+
+      break
+    }
+  }
+})
```

Comment:
Always using `{ force: true }` is problematic because you risk accidental passes of commands that shouldn’t pass because they’re _actually_ currently occluded (e.g. by a modal dialog) or invisible (e.g. hidden via `v-show` or something like that).

When interacting with UI components whose `input` elements are visually hidden so some custom UI can be built that goes beyond what, for example, `input[type="checkbox"]` offers, you basically have two options when `click()`ing them using Cypress:

- `cy.get('input').click({ force: true })`
- Get the associated label element and click that as it’s unlikely to ever be hidden

Now the second option is more complex because finding the `label` for an `input` again has two options: the label is a parent or the label is associated to the input with a `for` attribute matching the input’s `id` attribute. We do use both approaches in Kongponents and so would need to check for both.

Conditionally using `{ force: true }` based on a class name seems more reliable even though it could of course break if that class name changes. That however should be marked a breaking change and so shouldn’t pose a risk.

Reply:
We can’t add attributes to `label` elements that come via Kongponents components as usually attributes are bound to the `input` element (that’s a good thing). 

I’ve now combined the two step utilities and removed the `{ force: true }` from the select branch as it doesn’t need it.

Reply:
Why don't we put the data-testid on the parent `radio-button-group` and then `data-testid=['ingress-input-switch'] label`? Makes for a better test also I think as a human clicks on the label:

```vue
          <div class="radio-button-group" data-testid="ingress-input-switch">
            <KInputSwitch
              id="zone-ingress-enabled"
              v-model="zoneIngressEnabled"
              
            >
              <template #label>
                {{ i18n.t('zones.form.zoneIngressEnabledLabel') }}
              </template>
            </KInputSwitch>
          </div>
        </div>
```

Would something like that work? Means we can get rid of the classname check in the step also

Reply:
I didn’t try it because I don’t think it’s a better approach. It would rely on us keeping up this structure which I think is less likely than the components in Kongponents changing theirs.

Also I think determining the labels is probably the most generic and robust way to do this safe from having a dedicated utility for switches (and possibly other components with that implementation detail, not a fan of the sound of that).

I’m gonna leave this as it is now.

Reply:
Ok sounds good, thanks for working through it with me, appreciate the extra effort here.